### PR TITLE
Untangling Cleanup: Clean up `wpcom_is_nav_redesign_enabled()` flag

### PR DESCRIPTION
**Pull request by xavier-lc**
Related to: https://github.com/Automattic/dotcom-forge/issues/7927

## Proposed changes:

* As we've launched untangling & nav redesign, the `wpcom_is_nav_redesign_enabled` function name is not relevant anymore and can be confusing for future developers. 

This goes alongside this wpcom PR: D154312-code

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply this patch on your sandbox
* On a site using the classic admin interface style, set the `wpcom_classic_tour_completed` option to `'0'`:
  * On your sandbox, run `wpsh` and then `update_blog_option( SITE_ID, 'wpcom_classic_tour_completed', '0' );`
* Go to `/wp-admin/?admin-interface-changed=true`
* Check that you see the tour dialog:

image

### CODE REVIEW

**Review by fushar — COMMENTED** [review]
Thanks for this! Some comments:

1. One of the reasons we used `wpcom_is_nav_redesign_enabled()` (and therefore the massive `function_exists` checks) in wpcomsh, is so that we could quickly release/revert the feature in wpcomsh, without having to rely on Jetpack release cycle. Now that it's released, we don't need to rely on wpcomsh anymore. That's the reason why I proposed in the issue to just simply use `get_option( 'wpcom_admin_interface' ) === 'wp-admin'` as it's literally what it is.
2. If we still want to use functions, what I suggest is:
    - Put the function in another feature in mu-wpcom. E.g.: `admin-interface/admin-interface.php`
    - Use it and remove all the `function_exists` checks.
    - See this example https://github.com/Automattic/jetpack/pull/37993, where we added a similar `is_agency_managed_site()` function.
3. We should do similar thing in `wpcomsh` and not rely on the `feature-plugins/nav-redesign.php` file anymore, as that would defeat the purpose of the refactor.

What do you think?

**Comment by fushar** [comment]
Can we now rename the PR 😄

**Comment by xavier-lc** [comment]
> Can we now rename the PR 😄

@fushar haha right, done!

File: projects/plugins/wpcomsh/feature-plugins/nav-redesign.php [hunk 1]
```diff
@@ -5,17 +5,6 @@
  * @package wpcomsh
  */
 
-/**
- * Whether to enable the nav redesign.
- *
- * @return bool True if the nav redesign is enabled, false otherwise.
- */
-function wpcom_is_nav_redesign_enabled() {
```
Comment (fushar) [inline projects/plugins/wpcomsh/feature-plugins/nav-redesign.php]: Let's not remove it now, until all Jetpack changes are fully deployed.

**Review by fushar — APPROVED** [review]
Triple-checked the changes, and smoke-tested on a WoA dev test site and everything works as usual. Nice 👍

**Comment by fushar** [comment]
Merging this now to unblock things because we have another Untangling iteration this week!

**Comment by xavier-lc** [comment]
@fushar thanks for merging this!

**Pull request merged**